src/web/app/components/notification-banner/notification-banner.component.spec.ts

```diff
@@ -23,11 +59,90 @@ describe('NotificationBannerComponent', () => {
 
   beforeEach(() => {
     fixture = TestBed.createComponent(NotificationBannerComponent);
+    notificationService = TestBed.inject(NotificationService);
+    statusMessageService = TestBed.inject(StatusMessageService);
     component = fixture.componentInstance;
+    component.notificationTargetUser = NotificationTargetUser.STUDENT;
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should close after clicking X', () => {
+    jest.spyOn(notificationService, 'getUnreadNotificationsForTargetUser').mockReturnValue(of({
+      notifications: [testNotificationOne],
+    }));
+    component.ngOnInit();
+    fixture.detectChanges();
+
+    expect(component.isShown).toBeTruthy();
+    const button = fixture.debugElement.query(By.css('#btn-close-notif')).nativeElement;
+    button.click();
+    expect(component.isShown).toBeFalsy();
+  });
+
+  it('should close after clicking mark as read', () => {
+    jest.spyOn(notificationService, 'getUnreadNotificationsForTargetUser').mockReturnValue(of({
+      notifications: [testNotificationOne],
+    }));
+    jest.spyOn(notificationService, 'markNotificationAsRead').mockReturnValue(of({
+      readNotifications: [testNotificationOne.notificationId],
+    }));
+    const apiSpy: SpyInstance = jest.spyOn(notificationService, 'markNotificationAsRead')
+      .mockImplementation((request: MarkNotificationAsReadRequest) => {
+        expect(request.notificationId).toEqual(testNotificationOne.notificationId);
+        return of({
+          readNotifications: [request.notificationId],
+        });
+      });
+    const messageSpy: SpyInstance = jest.spyOn(statusMessageService, 'showSuccessToast')
+      .mockImplementation((args: string) => {
+        expect(args).toEqual('Notification marked as read.');
+      });
+    component.ngOnInit();
+    fixture.detectChanges();
+
+    expect(component.isShown).toBeTruthy();
+    const button = fixture.debugElement.query(By.css('#btn-mark-as-read')).nativeElement;
+    button.click();
+    expect(apiSpy).toBeCalledTimes(1);
+    expect(messageSpy).toBeCalledTimes(1);
+    expect(component.isShown).toBeFalsy();
+  });
+
+  it('should snap with default fields', () => {
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('should snap with no unread notifications', () => {
+    jest.spyOn(notificationService, 'getUnreadNotificationsForTargetUser').mockReturnValue(of({
+      notifications: [],
+    }));
+    component.ngOnInit();
+    expect(component.notifications.length).toEqual(0);
+    fixture.detectChanges();
+    expect(fixture).toMatchSnapshot();
```

Comment:
This is not the ideal way of snapshot testing. Here what you're doing is "have a starting state, do something, then the snapshot should be like this". What should be done is "have a state, then the snapshot should be like this".

Reply:
Thank you for the comments. I will try to improve it in the following commits.